### PR TITLE
Fixes #1467 - Send useful filter label item data through components to Management Toolbar

### LINKED ISSUE #1467: Missing information in filterLabelCloseClicked
When showing filter label items in the management toolbar, it's not really possible to implement a `handleFilterLabelCloseClicked` handler since [the fired event has no associated data](https://github.com/liferay/clay/blob/master/packages/clay-management-toolbar/src/ClayManagementToolbar.js#L133-L144) to know which filter was closed.

Additionally, one option that was mentioned by @adolfopa and @4lejandrito was that they could predict the urls beforehand, so we could simply make those links that do a standard navigation (or handle this globally for them)

### PR DESCRIPTION
**Pull request by carloslancha**

### CODE REVIEW

File: packages/clay-isomorphic/fixtures/expected/ClayFilterLabel.render [hunk 1]
```diff
@@ -1 +1 @@
-<li class="tbar-item"><div class="tbar-section"><span class="label label-dismissible component-label tbar-label"><span class="label-item label-item-expand"><div class="label-section">Category: Label 3</div></span><span class="label-item label-item-after"><button class="btn close"aria-label="close"type="button"><svg class="lexicon-icon lexicon-icon-times"focusable="false"role="presentation"><use xlink:href="icons.svg#times" /></svg></button></span></span></div></li>
\ No newline at end of file
+<div class="null"><span class="label label-dismissible component-label tbar-label"><span class="label-item label-item-expand"><div class="label-section">Category: Label 3</div></span><span class="label-item label-item-after"><button class="btn close"aria-label="close"type="button"><svg class="lexicon-icon lexicon-icon-times"focusable="false"role="presentation"><use xlink:href="icons.svg#times" /></svg></button></span></span></div>
```
Comment (jbalsas) [inline packages/clay-isomorphic/fixtures/expected/ClayFilterLabel.render]: This snapshot seems weird :D

File: packages/clay-isomorphic/fixtures/expected/ClayResultsBar.filterLabels [hunk 2]
```diff
@@ -1 +1 @@
...
\ No newline at end of file
+<li class="tbar-item"><div class="tbar-section"><div class="null"><span class="label label-dismissible component-label tbar-label"><span class="label-item label-item-expand"><div class="label-section">Label 1</div></span><span class="label-item label-item-after"><button class="btn close"aria-label="close"type="button"><svg class="lexicon-icon lexicon-icon-times"focusable="false"role="presentation"><use xlink:href="spritemap.svg#times" /></svg></button></span></span></div></div></li><li class="tbar-item tbar-item-expand"><div class="tbar-section"><div class="null"><span class="label label-dismissible component-label tbar-label"><span class="label-item label-item-expand"><div class="label-section">Label 2</div></span><span class="label-item label-item-after"><button class="btn close"aria-label="close"type="button"><svg class="lexicon-icon lexicon-icon-times"focusable="false"role="presentation"><use xlink:href="spritemap.svg#times" /></svg></button></span></span></div></div></li>
```
Comment (jbalsas) [inline packages/clay-isomorphic/fixtures/expected/ClayResultsBar.filterLabels]: Here too, see `div class="null"`

File: packages/clay-management-toolbar/src/ClayResultsBar.js [hunk 3]
```diff
@@ -37,6 +37,9 @@ class ClayResultsBar extends ClayComponent {
 	 */
 	_handleFilterLabelCloseClicked(event) {
 		return !this.emit({
+			data: {
+				label: this.filterLabels[event.target.data.labelId],
```
Comment (jbalsas) [inline packages/clay-management-toolbar/src/ClayResultsBar.js]: I think we will likely need more than the label value, as that can be localized or who know what else. We would be better off passing down the `id`, the whole list of `label` and any additional information we might need.

Comment (carloslancha) [reply packages/clay-management-toolbar/src/ClayResultsBar.js]: That is already the entire label object.

**Comment by carloslancha** [comment]
Already approved by @jbalsas. Merging.

**Pull request merged**